server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -525,6 +525,10 @@ protected Node(final Environment initialEnvironment, Collection<PluginInfo> clas
                 );
             }
 
+            // Ensure feature flags from opensearch.yml are valid during plugin initialization.
+            // Bug fix: https://github.com/opensearch-project/OpenSearch/issues/18417
```

Comment:
Nitpick, but the commit message/PR description is the right place for this type of context. Unless it's a TODO or some other exception case we don't need to be adding issue links for code modifications.